lib/active_merchant/billing/gateways/first_pay/first_pay_json.rb

```diff
@@ -0,0 +1,172 @@
+require 'active_merchant/billing/gateways/first_pay/first_pay_common'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class FirstPayJsonGateway < Gateway
+      include FirstPayCommon
+
+      ACTIONS = {
+        purchase: 'Sale',
+        authorize: 'Auth',
+        capture: 'Settle',
+        refund: 'Refund',
+        void: 'Void'
+      }.freeze
+
+      self.live_url = 'https://secure.1stpaygateway.net/secure/RestGW/Gateway/Transaction/'
+
+      # Creates a new FirstPayJsonGateway
+      #
+      # The gateway requires two values for connection to be passed
+      # in the +options+ hash.
+      #
+      # ==== Options
+      #
+      # * <tt>:merchant_key</tt>  -- FirstPay's merchant_key (REQUIRED)
+      # * <tt>:processor_id</tt>  -- FirstPay's processor_id or processorId (REQUIRED)
+      #   Otherwise, perform transactions against the production server.
+      def initialize(options = {})
+        requires!(options, :merchant_key, :processor_id)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+
+        commit(:purchase, post)
+      end
+
+      def authorize(money, payment, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+
+        commit(:authorize, post)
+      end
+
+      def capture(money, authorization, options = {})
```

Comment:
💬 Comment:

`options` parameter is not being used anywhere on `capture` nor `refund` methods